drivers/memc/memc_stm32_xspi_psram.c

```diff
@@ -14,11 +14,20 @@
 #include <zephyr/drivers/gpio.h>
 #include <zephyr/drivers/pinctrl.h>
 #include <zephyr/logging/log.h>
+#include <zephyr/multi_heap/shared_multi_heap.h>
 
 LOG_MODULE_REGISTER(memc_stm32_xspi_psram, CONFIG_MEMC_LOG_LEVEL);
 
 #define STM32_XSPI_NODE DT_INST_PARENT(0)
 
+#ifdef CONFIG_SHARED_MULTI_HEAP
+struct shared_multi_heap_region smh_psram = {
+	.addr = DT_REG_ADDR(DT_NODELABEL(psram)),
+	.size = DT_REG_SIZE(DT_NODELABEL(psram)),
+	.attr = SMH_REG_ATTR_EXTERNAL,
```

Comment:
A comparison with Espressif for pure curiosity:

https://github.com/zephyrproject-rtos/zephyr/blob/8ab712a2353862277fb23191a5f77e35057c3e63/soc/espressif/esp32s2/memory.h#L61-L62

https://github.com/zephyrproject-rtos/zephyr/blob/8ab712a2353862277fb23191a5f77e35057c3e63/soc/espressif/esp32/default.ld#L32-L35

https://github.com/zephyrproject-rtos/zephyr/blob/8ab712a2353862277fb23191a5f77e35057c3e63/soc/espressif/esp32/default.ld#L850-L853

https://github.com/zephyrproject-rtos/zephyr/blob/8ab712a2353862277fb23191a5f77e35057c3e63/soc/espressif/common/esp_psram.c#L17-L21